cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -282,6 +292,36 @@ func Run(ctx context.Context, c *config.CompletedConfig) error {
 			return startSATokenControllerInit(ctx, controllerContext, controllerName)
 		}
 	}
+	ver, err := semver.ParseTolerant(version.Get().String())
+	if err != nil {
+		return err
+	}
+
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.CoordinatedLeaderElection) {
+		// A separate informer factory is created because LeaseCandidate renewal
+		// must be performed regardless whether the current instance of KCM is
+		// leader or not. The normal KCM shared informer factory is started and
+		// shutdown based on leader election. LeaseCandidates also require a
+		// longer resyncInterval.
+		kubeInformerFactory := informers.NewSharedInformerFactory(c.Client, 5*time.Minute)
+		// Start component identity lease management
+		leaseCandidate, err := leaderelection.NewCandidate(
+			c.Client.CoordinationV1alpha1().LeaseCandidates("kube-system"),
+			kubeInformerFactory.Coordination().V1alpha1().LeaseCandidates(),
+			id,
+			"kube-system",
+			"kube-controller-manager",
+			clock.RealClock{},
+			ver.FinalizeVersion(),
+			ver.FinalizeVersion(), // TODO: Use compatibility version when it's available
+			[]v1.CoordinatedLeaseStrategy{"OldestEmulationVersion"},
+		)
+		if err != nil {
+			return err
+		}
+		go kubeInformerFactory.Start(ctx.Done())
+		go leaseCandidate.Run(ctx)
```

Comment:
Wait for sync?

Reply:
Also this should be part of readiness of the kcm process.

Reply:
Added as part of health check, I don't see a readiness check in kcm.